### PR TITLE
feat: LT-1127 - Control who can record the final decision

### PR DESCRIPTION
**Pull request by jfaracik**
* Views for getting all roles and permissions
* Views for adding and editing roles
* Permission class for certain roles

### CODE REVIEW

File: applications/views.py [hunk 1]
```diff
@@ -115,6 +117,12 @@ def put(self, request, pk):
         Update an application instance.
         """
         with reversion.create_revision():
+            request_data = json.loads(request.body)
+
+            if request_data.get('status') == 'approved' or request_data.get('status') == 'declined':
+                if not has_permission(request.user, Permissions.MAKE_FINAL_DECISIONS):
+                    return JsonResponse(data={'application': 'Permission not found'}, status=403)
```
Comment (markm5A8Z9F43913A) [inline applications/views.py]: Minimise the number of returns in each method.

Comment (cstobart) [reply applications/views.py]: each return is distinct so needs to be kept separate

File: gov_users/views/roles_views.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    def get(self, request, pk):
+        """
+        Get the details of a specific role
+        """
+        role = get_role_by_pk(pk)
+        serializer = RoleSerializer(role)
+
+        return JsonResponse(data={'role': serializer.data})
+
+    def put(self, request, pk):
```
Comment (lukesa) [inline gov_users/views/roles_views.py]: need to add swagger decorator to bind RoleSerializer

File: teams/serializers.py [hunk 3]
```diff
@@ -5,18 +5,14 @@
...
 class TeamSerializer(serializers.ModelSerializer):
-    name = serializers.CharField(
-        max_length=50,
-        validators=[UniqueValidator(queryset=Team.objects.all(), lookup='iexact',
-                                    message='Enter a name which is not already in use by another team')],
-        error_messages={'blank': 'Team name may not be blank'})
+    name = serializers.CharField(max_length=50,
+                                 validators=[
+                                     UniqueValidator(queryset=Team.objects.all(), lookup='iexact',
+                                                     message='Enter a name which is not already in use by another team')
```
Comment (lukesa) [inline teams/serializers.py]: get_strings

File: teams/serializers.py [hunk 4]
```diff
@@ -5,18 +5,14 @@
...
+                                 ],
+                                 error_messages={'blank': 'Team name may not be blank'})
```
Comment (lukesa) [inline teams/serializers.py]: get_strings

**Pull request merged**